File: src/main/java/io/bindernews/thegrackle/cards/AerialAdvantage.java
package io.bindernews.thegrackle.cards;

import com.megacrit.cardcrawl.actions.common.DrawCardAction;
import com.megacrit.cardcrawl.core.AbstractCreature;
import io.bindernews.bnsts.CardVariables;
import io.bindernews.thegrackle.stance.StanceAloft;
import io.bindernews.thegrackle.variables.Magic2Var;

import static io.bindernews.thegrackle.helper.ModInterop.iop;

public class AerialAdvantage extends BaseCard {
    public static final CardConfig C =
            new CardConfig("AerialAdvantage", CardType.SKILL, CardRarity.UNCOMMON, CardTarget.SELF);
    static final CardVariables VARS = CardVariables.config(c -> {
        c.cost(0);
        c.magic(2, 3);
        c.add(Magic2Var.inst, 4, -1);
        c.exhaust(true, true);
    });

    public AerialAdvantage() {
        super(C, VARS);
    }

    @Override
    public void apply(AbstractCreature p, AbstractCreature m) {
        // N.B Not applicable for charboss
        addToBot(iop().changeStance(p, StanceAloft.STANCE_ID));
        addToBot(iop().actionGainEnergy(p, magicNumber));
        addToBot(new DrawCardAction(p, Magic2Var.inst.value(this)));
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/cards/AerialAdvantage.kt
package io.bindernews.thegrackle.cards

import basemod.cardmods.ExhaustMod
import com.megacrit.cardcrawl.actions.common.DrawCardAction
import com.megacrit.cardcrawl.core.AbstractCreature
import io.bindernews.bnsts.CardVariables
import io.bindernews.thegrackle.helper.ModInterop.Companion.iop
import io.bindernews.thegrackle.helper.magic2
import io.bindernews.thegrackle.stance.StanceAloft

class AerialAdvantage : BaseCard(C, VARS) {
    override fun apply(p: AbstractCreature, m: AbstractCreature?) {
        // N.B Not applicable for charboss
        addToBot(iop().changeStance(p, StanceAloft.STANCE_ID))
        addToBot(iop().actionGainEnergy(p, magicNumber))
        addToBot(DrawCardAction(p, magic2))
    }

    companion object {
        val C = CardConfig("AerialAdvantage", CardType.SKILL, CardRarity.UNCOMMON, CardTarget.SELF)
        val VARS = CardVariables.config { c ->
            c.cost(0)
            c.magic(2, 3)
            c.magic2(4, -1)
            c.addModifier(ExhaustMod())
        }
    }
}